SingularityService/src/main/java/com/hubspot/singularity/auth/SingularityGroupsScopesAuthorizer.java

```diff
@@ -315,17 +270,50 @@ private void checkForbiddenForGroups(
     );
   }
 
+  private Set<String> getGroups(
+    SingularityRequest request,
+    SingularityAuthorizationScope scope
+  ) {
+    switch (scope) {
+      case READ:
+        return getReadGroups(request);
+      case WRITE:
+        return getWriteGroups(request);
+      case ADMIN:
+      default:
+        return Collections.emptySet();
+    }
+  }
+
   private Set<String> getReadGroups(SingularityRequest request) {
     Set<String> allowedReadGroups = new HashSet<>(
       authConfiguration.getGlobalReadOnlyGroups()
     );
     allowedReadGroups.addAll(authConfiguration.getGlobalReadWriteGroups());
-    if (!request.getReadOnlyGroups().isPresent()) {
-      allowedReadGroups.addAll(authConfiguration.getDefaultReadOnlyGroups());
+
+    if (request.getActionPermissions().isPresent()) {
+      request
+        .getActionPermissions()
```

Comment:
is this needed here still? I think the action permissions are all write-based (and also not SingularityAuthorizationScope objects)

Reply:
Good catch - missed this one